File: php-photos/deletephoto.php
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

    <head>
        <title>Photo Gallery</title>
        <link rel="stylesheet" type="text/css" href="res/css/Site.css">
        <link href="res/lightbox/css/lightbox.css" rel="stylesheet" />
        <script src="res/lightbox/js/jquery-1.10.2.min.js"></script>
        <script src="res/lightbox/js/lightbox-2.6.min.js"></script>
    </head>

    <body>

        <div id="pageOuter">

            <h1>PHP - Photo Gallery</h1>

			<?php include 'dbaccess.php'; ?>
			<?php 
				$photoId = 0;
				$confirm = 0;

				$photoId = $_GET["oid"];
				$confirm = $_GET["confirm"];

				if ($photoId <= 0)
				{
					echo "No object ID provided.";
				}
				else
				{
				try {
						$pdo = new PDO('mysql:host='.constant("host").';dbname='.constant("dbname"), 
											constant("user"), constant("password"));
						$qryResult = $pdo->query('SELECT * FROM tbl_photos WHERE id = '.$photoId);
						
						for($i=0; $row = $qryResult->fetch(); $i++){
							
							$photoId = $row['id'];
							$description = $row['description'];
							$fileRef = $row['file_ref'];
							$title = $row['title'];
							
							if ($confirm != 1)
							{
								echo "<p>Are you sure you want to delete this?<p>
									<a href='res/photos/$fileRef' rel='lightbox' title='$description'>
										<img class='photo' src='res/photos/$fileRef' data-lightbox='image-$photoId' />
									</a>
									<a href='index.php'>Cancel</a>&nbsp;&nbsp;
									<a href='?confirm=1&oid=$photoId'>Delete</a>";
							}
							//TODO: write a delete query in php perhaps using the query->prepare() method.
							/*
							else
							{
								try
								{
									echo "DELETED " + RMySqlDbHelper.deletePhoto(photoId).intResult + "PHOTO.";
									echo "<p><a href='Default.aspx'>Back to Gallery<a></p>";
								}
								catch (Exception ex) { output.InnerHtml += "DELETE ERROR: "; }
							}
							*/
							
						}
						$pdo = null;
				} catch (PDOException $e) {
						print "Error!: " . $e->getMessage() . "<br/>";
						die();
				}
			?>
			
		</div>
		
	</body>
	
</html>

File: pl-photos/htdocs/deletephoto.pl
#!/usr/bin/perl

use CGI::Carp qw(fatalsToBrowser);
use CGI qw(:standard);
use DBI;

print "Content-type: text/html\n\n";

$photoId = 0;
$confirm = 0;

#Get the photo ID and 'confirm' value from the posted param name
for $paramName (param()) {
    if (index($paramName, "delete") eq 0) {
        my @parts = split(/:/,$paramName);
        $photoId = $parts[1];
    }
    if (index($paramName, "confirmdelete") eq 0) {
        @parts = split(/:/,$paramName);
        $confirm = $parts[1];
    }
}

#If no photo ID then redirect back to home.
if ($photoId <= 0) {
    print redirect(-url => 'home.pl');
}

#Get the photo record from the database
$db = DBI->connect('dbi:mysql:photogallery1','root','GiaaHlm4e')
   or die "Connection Error: $DBI::errstr\n";
$sql = "SELECT id, title, description, file_ref FROM photos WHERE id = ? LIMIT 1";
my ($photoId, $photoTitle, $photoDesc, $photoFile) 
    = $db->selectrow_array($sql, undef, ($photoId));

my $content = "";

if ($photoId <= 0){
   $content = "Photo not found";
}
else {
    if($confirm == 1){
        #Delete the record from the DB
        $qDelete = $db->prepare("DELETE FROM photos WHERE id = ?");
        $qDelete->execute($photoId)
            or die "SQL Error: $DBI::errstr\n";
        $content = qq{
            <p>The photo has been deleted.</p>
            <p><a href='home.pl'><< Back to home page</a></p>
        }
    }
    else{
        #Ask user to confirm the deletion
        $content = qq{
            <p>Are you sure you want to delete this?<p>
            <a href='res/photos/$photoFile' rel='lightbox' title='$photoDesc'>
                <img class='photo' src='res/photos/$photoFile' data-lightbox='image-$photoId' />
            </a>
            <a href='home.pl'>Cancel</a>
            <form action='deletephoto.pl' method='post'>
                <input type='hidden' name='delete:$photoId' value='$photoId' /> 
                <input class='confirmDeletePhoto' type='submit' name='confirmdelete:1' value='Delete' /> 
            </form>
        };
    }
}

$html = qq{
    
    <html>
        <head>
            <title>Photos</title>
            <link rel="stylesheet" type="text/css" href="res/css/site.css">
        </head>
        <body>
            <div class="pageOuter">
                <h1>Photo App (Perl)</h1>
                <div id="containerPhotos">
                    $content
                </div>
            </div>
        </body>
    </html>

};

print $html;
